Zend/zend_ast.c

```diff
@@ -914,6 +893,34 @@ ZEND_API zend_result ZEND_FASTCALL zend_ast_evaluate(zval *result, zend_ast *ast
 	return ret;
 }
 
+ZEND_API zend_result ZEND_FASTCALL zend_ast_evaluate(zval *result, zend_ast *ast, zend_class_entry *scope)
+{
+	zend_string *previous_filename;
+	zend_long previous_lineno;
+	zend_result r;
+	bool bailout = false;
+
+	if (scope) {
+		previous_filename = EG(filename_override);
+		previous_lineno = EG(lineno_override);
+		EG(filename_override) = scope->info.user.filename;
+		EG(lineno_override) = zend_ast_get_lineno(ast);
+	}
+	zend_try {
+		r = zend_ast_evaluate_ex(result, ast, scope);
+	} zend_catch {
+		bailout = true;
+	} zend_end_try();
```

Comment:
Is it really necessary to catch fatal errors here?
We may reset EG(filename_override)/EG(lineno_override) at the end of request or after emitting a fatal error message.

Reply:
You're right. I wasn't sure if bailout can ever be caught during the request but it doesn't seem to be the case. I removed the try/catch.